### DISCUSSION TITLE
E2E Testing Instructions

### DISCUSSION THREAD

**Discussion by michaellatman** [open-olive/loop-development-kit#90] [Q&A]
Does Olive Helps have best practices or documentation on how to E2E test Olive loops? I'd like to run suites of automated tests before reaching the deployment stage of a pipeline.

**Comment by BrettMugglinOlive** [comment]
We haven't yet established what best practices are for automated E2E testing for ourselves yet actually! The complication comes from the Olive Helps platform itself which uses a different flavor of Electron, astilectron. It doesn't play nice with Electron's recommended CI tool, and we have been looking into ways to make one for ourselves.

In lieu of this, while it wouldn't be automated, you could repurpose what we have in our `self-test-loop` example, which a sort of interactive testing framework that we use to confirm certain systems are running correctly before we cut releases. To do this you would need to reconfigure the `testConfig` object to run the tests that you would like to execute, and implement those testing functions in `tests/index.ts`.

**Comment by cshields-oliveai** [comment]
@michaellatman - In order to ensure we are providing quick and accurate answers, we have decided to deactivate Github Discussions for this repository. I will lock this conversation in the meantime.

If what @BrettMugglinOlive recommended above doesn't solve your issue and you are still interested in this feature, we encourage you to open a feature request in the "Issues" section for this repository.

Thanks for providing ideas for enhancements for the LDK!
